src/backends/interflop-cancellation/interflop_cancellation.c

```diff
@@ -196,6 +226,11 @@ static void init_context(t_context *ctx) {
   ctx->seed = 0ULL;
 }
 
+void _interflop_finalize(__attribute__((unused)) void *context) {
+  if (mca_data)
+    free(mca_data);
```

Comment:
Instead of allocation mca_data dynamically, you could allocate it statically and avoid having to do this free().